test/p1LiquidatorProxy.test.ts

```diff
@@ -0,0 +1,245 @@
+import BigNumber from 'bignumber.js';
+import _ from 'lodash';
+
+import initializePerpetual from './helpers/initializePerpetual';
+import { expectBalances, mintAndDeposit } from './helpers/balances';
+import perpetualDescribe, { ITestContext } from './helpers/perpetualDescribe';
+import { buy } from './helpers/trade';
+import { expect, expectBN, expectThrow } from './helpers/Expect';
+import {
+  BigNumberable,
+  Price,
+  TxResult,
+  address,
+} from '../src/lib/types';
+
+const initialPrice = new Price(100);
+const longUndercollateralizedPrice = new Price(52);
+const shortUndercollateralizedPrice = new Price(140);
+const positionSize = new BigNumber(100);
+
+let admin: address;
+let long: address;
+let short: address;
+let neutral: address;
+let insuranceFund: address;
+let rando: address;
+
+interface LiquidateOptions {
+  liquidator: address;
+  liquidatee: address;
+  isBuy: boolean;
+  maxPosition: BigNumberable;
+}
+
+interface ExpectedLogOptions {
+  liquidator: address;
+  liquidatee: address;
+  isBuy: boolean;
+  feeAmount: BigNumberable;
+  liquidationAmount: BigNumberable;
+}
+
+async function init(ctx: ITestContext): Promise<void> {
+  await initializePerpetual(ctx);
+  admin = ctx.accounts[0];
+  long = ctx.accounts[1];
+  short = ctx.accounts[2];
+  neutral = ctx.accounts[3];
+  insuranceFund = ctx.accounts[4];
+  rando = ctx.accounts[5];
+
+  // Set up initial balances:
+  // +---------+--------+----------+-------------------+
+  // | account | margin | position | collateralization |
+  // |---------+--------+----------+-------------------|
+  // | long    |  -5000 |      100 |              200% |
+  // | short   |  15000 |     -100 |              150% |
+  // | netral  |  20000 |        0 |              INF% |
+  // +---------+--------+----------+-------------------+
+  await Promise.all([
+    ctx.perpetual.liquidatorProxy.setAllowance(),
```

Comment:
I thought this is called in deploy, I actually thought it would revert if called more than once due to the safeerc20 checks 🤔

Reply:
Only called in deploy for non-testnet

Reply:
Didn't realize that about SafeERC20. From a practical perspective it shouldn't matter, but let me look more into this

Reply:
No need to use SafeERC20 here, will use normal IERC20